builder/hyperv/iso/builder.go

```diff
@@ -0,0 +1,527 @@
+// Copyright (c) Microsoft Open Technologies, Inc.
+// All Rights Reserved.
+// Licensed under the Apache License, Version 2.0.
+// See License.txt in the project root for license information.
+package iso
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/mitchellh/multistep"
+	hypervcommon "github.com/mitchellh/packer/builder/hyperv/common"
+	"github.com/mitchellh/packer/common"
+	"github.com/mitchellh/packer/helper/communicator"
+	"github.com/mitchellh/packer/helper/config"
+	"github.com/mitchellh/packer/packer"
+	powershell "github.com/mitchellh/packer/powershell"
+	"github.com/mitchellh/packer/powershell/hyperv"
+	"github.com/mitchellh/packer/template/interpolate"
+)
+
+const (
+	DefaultDiskSize = 40 * 1024        // ~40GB
+	MinDiskSize     = 256              // 256MB
+	MaxDiskSize     = 64 * 1024 * 1024 // 64TB
+
+	DefaultRamSize                 = 1 * 1024  // 1GB
+	MinRamSize                     = 32        // 32MB
+	MaxRamSize                     = 32 * 1024 // 32GB
+	MinNestedVirtualizationRamSize = 4 * 1024  // 4GB
+
+	LowRam = 256 // 256MB
+
+	DefaultUsername = "vagrant"
+	DefaultPassword = "vagrant"
+)
+
+// Builder implements packer.Builder and builds the actual Hyperv
+// images.
+type Builder struct {
+	config Config
+	runner multistep.Runner
+}
+
+type Config struct {
+	common.PackerConfig         `mapstructure:",squash"`
+	common.HTTPConfig           `mapstructure:",squash"`
+	common.ISOConfig            `mapstructure:",squash"`
+	hypervcommon.FloppyConfig   `mapstructure:",squash"`
+	hypervcommon.OutputConfig   `mapstructure:",squash"`
+	hypervcommon.SSHConfig      `mapstructure:",squash"`
+	hypervcommon.RunConfig      `mapstructure:",squash"`
+	hypervcommon.ShutdownConfig `mapstructure:",squash"`
+
+	// The size, in megabytes, of the hard disk to create for the VM.
+	// By default, this is 130048 (about 127 GB).
+	DiskSize uint `mapstructure:"disk_size"`
+	// The size, in megabytes, of the computer memory in the VM.
+	// By default, this is 1024 (about 1 GB).
+	RamSizeMB uint `mapstructure:"ram_size_mb"`
```

Comment:
Having _mb_ in the _option_ name seems weird unless it cleanly maps to an api/command.